docs/topics/arrays.md

```diff
@@ -31,32 +50,244 @@ fun main() {
+|[`FloatArray`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-float-array/)|[`IntArray`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-int-array/)|[`LongArray`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-long-array/)|[`ShortArray`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-short-array/)|
+
+These classes have no inheritance relation to the `Array` class, but they
+have the same set of methods and properties. 
+
+```kotlin
+// Array of Int of size 5 with values [0, 0, 0, 0, 0]
+val exampleArray = IntArray(5)
+```
+
+> To convert primitive arrays to typed arrays, use [`toTypedArray()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/to-typed-array.html).
+>
+{type="note"}
+
+### Multidimensional arrays
+
+Arrays can be nested within each other to create multidimensional arrays:
+
+```kotlin
+fun main() {
+//sampleStart
+    // Creates a two-dimensional array
+    val twoDArray = Array(2) { IntArray(2) }
+    println(twoDArray.contentDeepToString())
+
+    // Creates a three-dimensional array
+    val threeDArray = Array(3) { Array(3) { IntArray(3)} }
+    println(threeDArray.contentDeepToString())
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="multidimensional-array-kotlin"}
+
+> Nested arrays don't have to be the same type or the same size.
+>
+{type="note"}
+
+## Access and modify elements
+
+Square brackets `[]` are transformed into `get()` or `set()` function calls, like in Java. For example, `x[a,b]` instead of `x.get(a,b)`:
+
+```kotlin
+fun main() {
+//sampleStart
+    val simpleArray = arrayOf(1, 2, 3)
+    simpleArray[0] = 10
+    println(simpleArray[0].toString())
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="access-array-kotlin"}
 
 Arrays in Kotlin are _invariant_. This means that Kotlin does not let us assign an `Array<String>`
 to an `Array<Any>`, which prevents a possible runtime failure (but you can use `Array<out Any>`,
 see [Type Projections](generics.md#type-projections)).
 
-## Primitive type arrays
+## Compare arrays
+
+To compare whether two arrays have the same elements in the same order, use [`contentEquals()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/content-equals.html).
+
+```kotlin
+fun main() {
+//sampleStart
+    val simpleArray = arrayOf(1, 2, 3)
+    val anotherArray = arrayOf(1, 2, 3)
+    println(simpleArray.contentEquals(anotherArray))
+
+    simpleArray[0] = 10
+    println(simpleArray.contentEquals(anotherArray))
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="compare-array-kotlin"}
+
+> Don't use equality (`==`) and inequality (`!=`) [operators](equality.md#structural-equality) to compare the contents of arrays. These operators
```

Comment:
Another big drawback of using Arrays, might be included in the discussion in the beginning (see my first comment)